pkg/daemon/ceph/client/pool.go

```diff
@@ -215,6 +216,28 @@ func CreatePoolForApp(context *clusterd.Context, clusterName string, newPool Cep
 	return nil
 }
 
+func createReplicationCrushRule(context *clusterd.Context, clusterName string, newPool CephStoragePoolDetails, ruleName string) error {
+	if newPool.FailureDomain == "" {
+		return nil
+	}
+
+	// set the crush root to the default if not already specified
+	var crushRoot string
+	if newPool.CrushRoot != "" {
+		crushRoot = newPool.CrushRoot
+	} else {
+		crushRoot = "default"
+	}
+
+	args := []string{"osd", "crush", "rule", "create-simple", ruleName, crushRoot, newPool.FailureDomain}
+	_, err := ExecuteCephCommand(context, clusterName, args)
+	if err != nil {
+		return fmt.Errorf("failed to crush rule %s. %+v", newPool.Name, err)
```

Comment:
the `ruleName` var should probably be used in this error message instead

Reply:
yes, good catch